server/routers/share-recovery/generateWalletRecoveryChallenge.ts

```diff
@@ -19,37 +19,61 @@ export const generateWalletRecoveryChallenge = protectedProcedure
     // operation will probably reuse it. Otherwise, the cleanup cronjobs will take care of it:
     const deviceAndLocationIdPromise = getDeviceAndLocationId(ctx);
 
-    const userWallet = await ctx.prisma.wallet.findFirst({
-      select: { id: true, status: true },
+    console.log(`Generating wallet recovery challenge for wallet: ${input.walletId}, user: ${ctx.user.id}`);
+
+    // Explicitly set JWT claims before wallet lookup
+    try {
+      // Set JWT claims with session-level SET
+      const jwtClaims = JSON.stringify({
```

Comment:
Based on the latest comments, this is already included in `ensureJwtClaimsSet`, right?